sphinx_immaterial/mermaid_diagrams.py

```diff
@@ -34,6 +40,7 @@ def run(self) -> List[nodes.Node]:
             self.add_name(diagram_div)
         diagram_div += diagram
         self.set_source_info(diagram_div)
+        setattr(self.env, _COPY_MERMAID_DIST_ENV_KEY, True)
```

Comment:
To support parallel building, a handler for `env-merge-info` is needed so that this information gets propagated back to the main process.  Otherwise, it will only be set in the workers that happen to process documents with mermaid directives.

https://www.sphinx-doc.org/en/master/extdev/appapi.html#event-env-merge-info

Reply:
I think I did this as requested. I basically took a hint from the inline_icons.py module:
https://github.com/jbms/sphinx-immaterial/blob/6713712c7211b7c684b7be271aebd8559941e1af/sphinx_immaterial/inline_icons.py#L80-L90